### PR TITLE
feat(server-info): add dash alias and multi-page for roles

### PR DESCRIPTION
**Pull request by kyranet**
- feat(server-info): add dashed aliases
- feat(server-info): add paginated message for roles
- feat(role-info): add dashed aliases

### CODE REVIEW

**Review by favna — DISMISSED** [review]

File: src/commands/Management/Helpers/guild-info.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	private getRoles(args: SkyraCommand.Args): Role[] {
+		const roles = [...args.message.guild!.roles.cache.values()].sort(SORT);
+		// Pop off the @everyone role
+		roles.pop();
+
+		return roles;
+	}
+
+	private getSummaryRoles(args: SkyraCommand.Args, roles: Role[]): string {
+		if (roles.length <= roleLimit) return args.t(LanguageKeys.Globals.AndListValue, { value: roles.map((role) => role.toString()) });
```
Comment (vladfrangu) [inline src/commands/Management/Helpers/guild-info.ts]: Use roleMention function
```suggestion
		if (roles.length <= roleLimit) return args.t(LanguageKeys.Globals.AndListValue, { value: roles.map(roleMention) });
```

File: src/commands/Management/Helpers/guild-info.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+		const mentions = roles
+			.slice(0, roleLimit - 1)
+			.map((role) => role.toString())
```
Comment (vladfrangu) [inline src/commands/Management/Helpers/guild-info.ts]: ^
```suggestion
			.map(roleMention)
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

File: src/commands/Management/Helpers/guild-info.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+		const display = new UserPaginatedMessage({
+			template: new MessageEmbed() //
+				.setColor(color)
+				.setThumbnail(guild.iconURL()!)
+				.setTitle(`${guild.name} [${guild.id}]`)
+		});
+
+		display.addPageEmbed(await this.getSummary(args, roles, color));
+
+		if (roles.length <= roleLimit) return display;
```
Comment (vladfrangu) [inline src/commands/Management/Helpers/guild-info.ts:47]: So wait.. if there's less than 15 roles, you enumerate them, else you... enumerate AND list fully?

Comment (kyranet) [reply src/commands/Management/Helpers/guild-info.ts:47]: Enumerate fully... in other pages. It's just useless to generate role pages if they're all in the summary.

File: src/commands/Management/Helpers/guild-info.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import { MessageEmbed, Permissions, Role } from 'discord.js';
+
+const SORT = (x: Role, y: Role) => Number(y.position > x.position) || Number(x.position === y.position) - 1;
+const roleMention = (role: Role) => role.toString();
+const roleLimit = 15;
+
+const paginatedMessagePermissions = new Permissions([Permissions.FLAGS.ADD_REACTIONS, Permissions.FLAGS.MANAGE_MESSAGES]);
+
+@ApplyOptions<SkyraCommand.Options>({
+	aliases: ['server-info'],
```
Comment (vladfrangu) [inline src/commands/Management/Helpers/guild-info.ts:16]: add dashless in here too (`generateDashLessAliases`)

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/commands/Management/Helpers/guild-info.ts (hunk 4 above)
Comment (kyranet) [reply src/commands/Management/Helpers/guild-info.ts:16]: Dashless aliases are auto-generated by default.

https://github.com/skyra-project/skyra/blob/4c1e5765160ae0c866101d74cec9d94ed32f3533/src/lib/structures/commands/SkyraCommand.ts#L75

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**